src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -68,6 +68,7 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
             GrafanaSpec, self.mgr.spec_store.active_specs[daemon_spec.service_name])
         grafana_ini = self.mgr.template.render(
             'services/grafana/grafana.ini.j2', {
+                'anonymous_access': spec.anonymous_access,
```

Comment:
BTW, in an upgrade scenario: would the upgrade fail because the old spec does't have this new field? .. in that case probably we might check first check if the field is there  otherwise pass false!

Reply:
I think when the mgr first comes up on the new version and loads the spec from the json we store in the k-v store the field will start to exist and that will happen before we ever run this code.

Reply:
Thanks for the clarification.